common/lockmanager_test.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edulinq/autograder/config"
+)
+
+var (
+    key = "testkey1"
+    key2 = "testkey2"
+    doesNotExistKey = "dne"
+    staleDuration = time.Duration(config.STALELOCK_DURATION.Get()) * time.Second;
+    wg sync.WaitGroup;
+)
+
+func TestLockManager(test *testing.T) {
+    testLockingKey()
+    
+    testUnlockingKey(test)
+
+    testConcurrentLockingUnlocking()
+
+    testUnlockingAnUnlockedLock(test)
+    
+    testUnlockingMissingKey(test)
+
+    testLockConcurrencyWithStaleCheck(test)
+}
+
+func testLockingKey() {
+    Lock(key)
+}
+
+func testUnlockingKey(test *testing.T) {
+    err := Unlock(key);
+    if (err != nil) {
+        test.Errorf("Failed to unlock a key");
+    }
+}
+
+func testConcurrentLockingUnlocking() {
+    Lock(key);
+    Lock(key2);
+    wg.Add(2);
+    go func() {
+        defer wg.Done();
+        Lock(key);
+        defer Unlock(key);
+    }()
+    go func() {
+        defer wg.Done();
+        Lock(key2);
+        defer Unlock(key2);
+    }();
+    Unlock(key);
+    Unlock(key2);
+
+    wg.Wait();
+}
+
```

Comment:
I'm going to stop reviewing this file.
I think my comments above in this file will give you a good start and restructuring this and I think future comment will probably be repetitive.